pkg/sentry/sentry.go

```diff
@@ -104,60 +111,96 @@ func (s *sentry) Start(ctx context.Context) error {
 		return fmt.Errorf("error creating security: %s", err)
 	}
 
-	val, err := s.validator(ctx)
+	vals, err := s.getValidators(ctx)
 	if err != nil {
 		return err
 	}
 
-	validatorErr := make(chan error, 1)
-	go func() {
-		log.Info("Starting validator")
-		validatorErr <- val.Start(ctx)
-	}()
-
-	log.Info("Validator created, starting sentry server")
+	// Start all background processes
+	runners := []concurrency.Runner{
+		provider.Start,
+		func(ctx context.Context) error {
+			sec, err := provider.Handler(ctx)
+			if err != nil {
+				return err
+			}
 
-	providerErr := make(chan error, 1)
-	go func() {
-		log.Info("starting security provider")
-		providerErr <- provider.Start(ctx)
-	}()
+			return server.Start(ctx, server.Options{
+				Port:             s.conf.Port,
+				Security:         sec,
+				Validators:       vals,
+				DefaultValidator: s.conf.DefaultValidator,
+				CA:               camngr,
+			})
+		},
+	}
+	for name, val := range vals {
+		log.Infof("Starting validator %s", strings.ToLower(name.String()))
```

Comment:
Pedantically, we are not running any validator at this point.

```suggestion
		log.Infof("Using validator %q", strings.ToLower(name.String()))
```